platform/config/ensureDeepObject.ts

```diff
@@ -1,29 +1,42 @@
 const separator = '.';
 
+/**
+ * Recursively traverses through the object's properties and expands ones with
+ * dot-separated names into nested objects (eg. { a.b: 'c'} -> { a: { b: 'c' }).
+ * @param {*} obj Object to traverse through
+ * @returns {Object} Same object instance with expanded properties.
+ */
 export function ensureDeepObject(obj: any) {
   if (obj == null || typeof obj !== 'object') {
     return obj;
   }
 
-  const fullObject = {} as any;
-  Object.keys(obj).forEach(k => {
-    if (!k.includes(separator)) {
-      fullObject[k] = ensureDeepObject(obj[k]);
+  if (Array.isArray(obj)) {
+    return obj.map((item) => ensureDeepObject(item));
+  }
+
+  return Object.keys(obj).reduce((fullObject, propertyKey) => {
```

Comment:
I was thinking about `Array.from(Object.entries())` here instead of `Object.keys`, but it requires additional lib for `Typescript` (`es2017.object`) and not sure we want it.